automation.md

```diff
@@ -0,0 +1,102 @@
+- UIA (UI Automation) 
+UI Automation provides programmatic access to information about the user interface (UI) for Microsoft windows, enabling assistive technology products (such as screen readers) to provide information about the UI to end users using means other than standard input and Manipulate UI. UI Automation also enables automated test scripts to interact with the UI.
+- MSAA（Microsoft Active Accessibility）
+Microsoft Active Accessibility was the earlier application accessibility solution from Microsoft. 
+
+UIA is the new accessibility model for Microsoft Windows, offers many improvements over MSAA.
+
+||UIA|MSAA|
+|--------|---------|-------|
+|Programing Languages| written in managed code, client applications are most easily programmed using C# | based on the Component Object Model (COM) with support for dual interfaces.|
+|Windows Presentation Foundation|full support |WPF do not contain native support for MSAA|
+|Servers and Clients|a core service lies between the server (called a provider) and the client, expose more information to client|servers and clients communicate directly, largely through the server's implementation of IAccessible|
+|Security|UIA removes the need for providers to call through to other provider code. The UI Automation core service does all the necessary aggregation.|Some IAccessible customization scenarios require wrapping a base IAccessible and calling through to it. This has security implications, since a partially trusted component should not be an intermediary on a code path.|
+
+If you are interested in Windows automation technology, you can directly read [Microsoft's official technical documentation](https://docs.microsoft.com/zh-CN/dotnet/framework/ui-automation/ui-automation-overview).
+
+# Java Access Bridge (JAB)
+Java Access Bridge is a technology that exposes the Java Accessibility API in a Microsoft Windows DLL, enabling Java applications and applets that implement the Java Accessibility API to be visible to assistive technologies on Microsoft Windows systems. Java Accessibility API is part of Java Accessibility Utilities, which is a set of utility classes that help assistive technologies provide access to GUI toolkits that implement the Java Accessibility API.
+This is the architecture of Java Access Bridge from oracle official site:
+
+![jab architecture](img/jab-block-diagram.gif)
+
+To learn more about JAB, please visit [oracle](https://docs.oracle.com/javase/8/docs/technotes/guides/access/jab/introduction.html#jab-overview)
+
+# Web Automation
+Web automation mainly depends on the type of browser, currently there are two main categories:
+
+## Internet Explorer(IE)  
+
+Through javascript injection, the difficulty is that there are too many versions of IE, and it is still technically difficult to make multiple versions of IE compatible.
```

Comment:
Through javascript injection, due to different versions of IE,  it is still a technical difficulty of making multiple versions of IE compatible.